Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java

```diff
@@ -0,0 +1,152 @@
+package io.branch.referral;
+
+import android.content.Context;
+
+import org.json.JSONObject;
+
+/**
+ * <p>
+ * Class for creating builder for getting a short url with Branch. This builder provide an easy and flexible way to configure and create
+ * a short url Synchronously or asynchronously.
+ * </p>
+ */
+public class BranchShortLinkBuilder extends BranchUrlBuilder {
+    public BranchShortLinkBuilder(Context context) {
+        super(context);
+    }
+
+    /**
+     * <p>Adds a tag to the iterable collection of name associated with a deep link.</p>
+     *
+     * @param tag {@link String} tags associated with a deep link.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder addTag(String tag) {
+        super.addTag(tag);
+        return this;
+    }
+
+    /**
+     * <p>Adds the the given key value pair to the parameters associated with this link.</p>
+     *
+     * @param key   A {@link String} with value of key for the parameter
+     * @param value A {@link String} with value of value for the parameter
+     * @returnA {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder addParameters(String key, String value) {
```

Comment:
The advantage is strictly in terms of removing redundancy. From here, it appears you have three separate classes that contain ~90% of the same code. Is there no way to make them all inherit from a single class? Even if ShortUrl has it's own builder, maybe all three extend from an AbstractUrlBuilder class or something.